Python/samples/cartpole/multi-concept-selector.ink

```diff
@@ -0,0 +1,184 @@
+            cart_velocity: obs.cart_velocity,
+            pole_angle: obs.pole_angle,
+            pole_angular_velocity: obs.pole_angular_velocity,}
+}
+
+graph (input: ObservedState): Action {
+    concept ComputeDelta(input): LearningState {
+        programmed SimplifyState
+    }
+
+    concept GoLeft(ComputeDelta): Action {
+        curriculum {
+            source CartpoleSim
+            goal (state: SimState) {
+                avoid FallOver:
+                    Math.Abs(state.pole_angle) in Goal.RangeAbove(MaxPoleAngle)
+
+                # There isn't a specific "close enough" threshold, so using
+                # Goal.RangeBelow(0) to encourage the system to minimize as well as it can
+                minimize DistToTarget:
+                    Math.Abs(state.target_pole_position - state.cart_position) in Goal.RangeBelow(0)
+            }
+            training {
+                EpisodeIterationLimit: 200,
+            }
+            lesson One {
+                # Because we're using relative positions, just need to randomize initial position to be far and close to the target
+                scenario {
+                    initial_cart_position: number<0 .. MaxPosition>,
+                    target_pole_position: 0
+                }
+            }
+        }
+    }
+
+    concept GoRight(ComputeDelta): Action {
+        curriculum {
+            source CartpoleSim
+            goal (state: SimState) {
+                avoid FallOver:
+                    Math.Abs(state.pole_angle) in Goal.RangeAbove(MaxPoleAngle)
+
+                # There isn't a specific "close enough" threshold, so using
+                # Goal.RangeBelow(0) to encourage the system to minimize as well as it can
+                minimize DistToTarget:
+                    Math.Abs(state.target_pole_position - state.cart_position) in Goal.RangeBelow(0)
+            }
+            training {
+                EpisodeIterationLimit: 200,
+            }
+            lesson One {
+                # Because we're using relative positions, just need to randomize initial position to be far and close to the target
+                scenario {
+                    initial_cart_position: number<-MaxPosition .. 0>,
+                    target_pole_position: 0
+                }
+            }
+        }
+    }
+
+    output concept PickOne(ComputeDelta): Action {
+       select GoRight
+       select GoLeft
+       curriculum {
+            source CartpoleSim
+            training {
+                EpisodeIterationLimit: 200,
+            }
+            goal (state: SimState) {
+                avoid FallOver: Math.Abs(state.pole_angle) in Goal.RangeAbove(MaxPoleAngle)
+                minimize DistToTarget: Math.Abs(state.target_pole_position - state.cart_position) in Goal.RangeBelow(0)
+            }
+            lesson One {
+                # Because we're using relative positions, just need to randomize initial position to be far and close to the target
```

Comment:
Suggest updating these comments for each concept to point out the differences. GoLeft starts the cart to the right of the target, GoRight left of the target, and PickOne uses the full range...